### ISSUE TITLE
Crashlytics firebase

### ISSUE THREAD

**Issue by netstart** [firebase/firebase-js-sdk#710]
Please, support crashlytics firebase. 

I do not want to use other services, like https://www.bugsnag.com/pricing/ report crash in my web app writed in Angular

**Comment by vladimirdjurdjevic** [comment]
Any news on this? It's would be really helpful if we had crashlytics for the web applications now that we have performance monitoring supported.

**Comment by IchordeDionysos** [comment]
Seems like Crashlytics is now the only GA Firebase product not supported for web...

**Comment by vladimirdjurdjevic** [comment]
Yeah... It's really missing piece of the puzzle. Hope it will be considered in near future :)

**Comment by smashah** [comment]
Maybe this can be integrated into firebase-performance?

https://github.com/GoogleCloudPlatform/stackdriver-errors-js

**Comment by dzienisz** [comment]
any updates on that?

**Comment by IchordeDionysos** [comment]
@dzienisz I would just relax and wait until end of March they are currently rebuilding the Crashlytics SDK as the current one was built for Fabric (Crashlytics was a product of Fabric).

And maybe this will enable them to also support Crashlytics for Web.

**Comment by dan-ryan** [comment]
Haven't seen anything about the web in their new rebuild.

**Comment by derhuebiii** [comment]
+1

**Comment by jorgehclinhares** [comment]
+1

**Comment by nathgilson** [comment]
+1 😭

**Comment by jkevingutierrez** [comment]
+1

**Comment by shahidcodes** [comment]
+1

**Comment by akauppi** [comment]
I’m building a template for ES module Firebase apps. Fully agree with @netstart ’s original text that this should/could be within the Firebase umbrella.

**Comment by JulianSprung** [comment]
I was hoping to use Crashlytics for a react-native & expo mobile app without ever having to touch any android/iOS-specific code.
Looks like I have to use Bugsnag or is there any way to integrate Crashlytics (for web)?

**Comment by hsubox76** [comment]
For React Native, you could use React Native Firebase https://rnfirebase.io/ but it would have to be plain React Native, as it doesn't work with Expo. It doesn't require using any native mobile code though.

**Comment by JulianSprung** [comment]
Thanks @hsubox76 for the quick reply and the pointer. 
That is a path forward if I eject from Expo and start using "bare" React Native.

Just to clarify, the only way I could use Firebase Crashlytics in my mobile app that is running on Expo would be if Crahslytics supported JS/Web apps, right?

**Comment by hsubox76** [comment]
I'm not sure what form a Crashlytics for Web product would take, or what the underlying technology would be, so I can't guarantee it would work with React Native or Expo.

But as this is a feature request issue, it seems like a good place for users to register a preference for any solution to work for React Native or anyone else's preferred environment.

**Comment by carloshwa** [comment]
A Javascript solution would also allow support for Flutter Web. Currently, the official supported crash reporting tool for Flutter Web is Sentry, which works well but would be nice to integrate with Crashlytics since most mobile apps will likely use that.

**Comment by robokozo** [comment]
I'd love to see this added one day!

**Comment by kengleong** [comment]
+1

**Comment by gvillo** [comment]
+1

**Comment by krzcho** [comment]
> +1

Why can't you just mark exiting comment with a reaction and instead you put meaningless notification in dozens of mailboxes???

**Comment by tohagan** [comment]
While [various hacks exist](https://stackoverflow.com/questions/54322479/can-we-use-firebase-crashlytics-for-our-web-application) we're still hanging out for this important feature.  Azure app insights [supports web apps](https://docs.microsoft.com/en-us/azure/azure-monitor/app/javascript) which was first release in 2015!

**Comment by amr-eniou-3r** [comment]
Any updates?

**Comment by kamil-ignac** [comment]
still no updates? Or any ETA?

**Comment by arthabus** [comment]
In my case I ended up using google's stackdrive which allows to keep track of errors and manage them. Its free and it's still from google. You can check how to set it up in [this stackoverflow answer](https://stackoverflow.com/a/65500440/1713920). 

Works well so far while we are waiting for crashlytics support.

**Comment by shauryaaher** [comment]
> Seems like Crashlytics is now the only GA Firebase product not supported for web...

Nope. A/B Testing, Predictions and In-App messaging are still not supported for the web.

**Comment by Galti** [comment]
🚬 Still waiting. Any updates.
We want to use our exiting app on web, too but this stops us.

**Comment by akauppi** [comment]
@Galti Have you considered Sentry?  I've gone there.

**Comment by zs-dima** [comment]
@akauppi Sentry is Sentry and firebase is firebase. Current issue about firebase.
(however Sentry nice tool and more or less good workaround for now)

**Comment by Galti** [comment]
@akauppi Thank you for your suggestion. But sometimes you can't decide which library/service to use in your app as you are not the CTO/Team Lead.
By the way firebase works great for mobile and I think running it on the web is not enough reason to change to Sentry and do refactoring.

**Comment by arthabus** [comment]
There is another alternative for the time being - Stackdrive, which is also
google's product.

Here is a post on how to use it:
https://stackoverflow.com/a/65500440/1713920

**Comment by shauryaaher** [comment]
I wanna use Firebase Crashlytics for my app, nothing else. That's the only thing I know of that provides detailed and nice crash reports. Also, the Crashlytics SDK may have a small bundle size.

**Comment by RoarGronmo** [comment]
In a total perspective Firebase Crashlytics should focus on web-support, especieally when there isn't any emphasize on those pages that Crashlytics isn't supported for Flutter Web (and other web based systems), when it is supported for iOS, android.

If it is supported for Flutter Web (lately ?), please let is know. Page ref:

https://pub.dev/packages/firebase_crashlytics  : emphasizes that there is no web support!!
https://firebase.google.com/docs/crashlytics/get-started?platform=flutter : not mentioned

**Comment by tohagan** [comment]
Reading about the latest cool Crashlytics feature feels kinda like another Firebase "let down" if you're a web dev. Seems like the Crashlytics devs have a "don't bother us" attitude to the web SDK.  Seriously how hard can it be? It would seem like most of the stack is in place.

**Comment by arthabus** [comment]
While I'm looking forward to having crashlytics, I don't think whining
would do much to help.

If you need to track crashes, use Stackdrive for the time being - there is
nothing to integrate, just enable it in your google account and you'll get
crash reports. If you use standard "console.error" logs that's it, you
don't even need to change your code at all.

If Stackdrive is not an option, then just wait.

Or you can become an employee of the Crashlytics team and implement it.
Everyone else here would be grateful.

Other than that, comments with the whine attitude don't really add anything
to the conversation, so please stop posting them.

**Comment by hsubox76** [comment]
Speaking for the JS SDK team, we would be happy to add Crashlytics to web and we know the high demand for it. A lot of moving parts need to come together to make that happen, so we can't guarantee anything. But feature request issues like this are very important for demonstrating demand and getting those moving parts to start moving, etc, so if this is important to you, absolutely thumbs up the original post.

And if you have any interesting use cases or reasons why it's important to you, or other Google/Firebase products you would use more of if Crashlytics was available, please feel free to add comments about it.

Also welcome are what are the most important features or requirements of a Crashlytics Web SDK? It will necessarily be different from mobile crashlytics as the definition of crashes and errors aren't perfectly parallel between mobile and web platforms. Or if there are other existing products out there that you like and you want us to match.

**Comment by cbwhitebu** [comment]
Interested particularly in Crashlytics for Flutter. Would love to the get the Dart stack trace in error reports.

**Comment by kinex** [comment]
> Also welcome are what are the most important features or requirements of a Crashlytics Web SDK? It will necessarily be different from mobile crashlytics as the definition of crashes and errors aren't perfectly parallel between mobile and web platforms. Or if there are other existing products out there that you like and you want us to match.

For a Flutter app I don't see any differences between the definition of mobile/Web crashes and errors. 

My Flutter app is available for Android, iOS and Web (+PWA app). I have added Sentry to my Web app, but I don't really use it because decrypting minified stack traces seems to require uploading my source codes to Sentry backend which I don't want to do. Also adding another service like Sentry to my app is just extra work and extra risk regarding e.g. privacy policy.  So currently my Flutter app for Web is without functional error reporting. I fix issues I see from Android/iOS Crashlytics and trust that it is good enough also for Web for now. 

I am not that familiar with other Web development tools, but I believe Flutter/Web developers in particular miss Crashlytics for Web a lot.

**Comment by mdrideout** [comment]
> Also welcome are what are the most important features or requirements of a Crashlytics Web SDK? It will necessarily be different from mobile crashlytics as the definition of crashes and errors aren't perfectly parallel between mobile and web platforms. Or if there are other existing products out there that you like and you want us to match.

Our Flutter Web app could really use Crashlytics. Web is our primary deployment target, and occasionally Web has exceptions and issues that are not present on mobile, especially with federated plugin packages that have different implementation across platforms. Tracking web OAuth flows for exceptions in production, for example, is virtually impossible right now.

We are currently using Sentry instead of Crashlytics because of the multi-platform support - however, while Sentry allows us to see that flutter web exceptions are happening, there is a source map code minification issue https://github.com/flutter/flutter/issues/87734#issuecomment-927627149 that prevents accurate reporting of which part of the code is causing the exception.

**Comment by agrohs** [comment]
Any update on here - been waiting patiently for years....

**Comment by bombillazo** [comment]
Very much needed...

**Comment by risalfajar** [comment]
Welcome to 2023, still waiting

**Comment by jcamilo1599** [comment]
> Welcome to 2023, still waiting

And the wait will continue. It seems that it will not be taken into account.

**Comment by Logaritma26** [comment]
Its fun that looks like we'll close this year without any update on this lol

**Comment by ted-marozzi** [comment]
Sentry is actually really good lol just use it, don't worry about waiting

**Comment by sniper0110** [comment]
+1

**Comment by tohagan** [comment]
I found i can just activate Google Error Reporting, create a simple firebase function that accepts amd logs an Error object and call it from my web apps. By sending an Error object from the client, you ensure the client side stack is captured so the Error Reporting system can aggregate similar errors and allow you to manage and track resolutions.

https://cloud.google.com/error-reporting

**Comment by adnandrops** [comment]
> Speaking for the JS SDK team, we would be happy to add Crashlytics to web and we know the high demand for it. A lot of moving parts need to come together to make that happen, so we can't guarantee anything. But feature request issues like this are very important for demonstrating demand and getting those moving parts to start moving, etc, so if this is important to you, absolutely thumbs up the original post.
> 
> And if you have any interesting use cases or reasons why it's important to you, or other Google/Firebase products you would use more of if Crashlytics was available, please feel free to add comments about it.
> 
> Also welcome are what are the most important features or requirements of a Crashlytics Web SDK? It will necessarily be different from mobile crashlytics as the definition of crashes and errors aren't perfectly parallel between mobile and web platforms. Or if there are other existing products out there that you like and you want us to match.

hey, any news on the progress or timeline? 
Tell us anything please!

**Comment by expresspotato** [comment]
> Welcome to 2023, still waiting
> 
> UPDATE: To anyone still waiting for this FR, believe me, just move on to Sentry.
> 
> Even if the Firebase Team want to implement this, it'll take a long time until it gets as advanced as Sentry.

I've also switch to Sentry and am loving it compared to Google Firebase ...

**Comment by hamzament** [comment]
It's 2024, nothing new on this? does the flutter team even intend to support it?

**Comment by risalfajar** [comment]
> Welcome to 2023, still waiting
> 
> UPDATE: To anyone still waiting for this FR, believe me, just move on to Sentry.
> 
> Even if the Firebase Team want to implement this, it'll take a long time until it gets as advanced as Sentry.

Welcome to 2024, I'll say this again

**Comment by wjkoh** [comment]
I'm looking forward to it!

**Comment by lisovyk** [comment]
Astounded by how high the demand is yet and nothing changed in 6 (!!!) years.

**Comment by Matthew-Shannon** [comment]
@lisovyk  I was just thinking the same thing. Still no Crashlytics for web after 6 years. Any recommendations as to an alternative?

**Comment by itssidhere** [comment]
How come this has still not been implemented?

**Comment by nabil-hfz** [comment]
+1 would love this

**Comment by pasalino** [comment]
+1 would love this

**Comment by PasinduDG** [comment]
+1 would be great to have this

**Comment by epsyan** [comment]
+1 me and my team would be superhappy

**Comment by alexandre1921** [comment]
+1 me and also my team would be happy too

**Comment by pasza01** [comment]
Need this feature in js sdk

**Comment by ThirdActJonas** [comment]
+1, would really love to have all my app's crash-data in one service

**Comment by roscoekerby** [comment]
+1 please, I have been struggling with this all morning... I thought I was going crazy...

**Comment by r1al12** [comment]
+1, good feature to have.

**Comment by maremoto** [comment]
+1 Yes, Please!

**Comment by l1qu1d** [comment]
This is ***by far*** the most requested feature `is:issue state:open label:"feature request" reactions:>=100 `.  There needs to be an answer (either to implement or not), timeline, and/or roadmap by now.

**Comment by pachesnyi** [comment]
+1, would be nice to have this one

**Comment by LorenzoVaccher01** [comment]
+1 💪🏻

**Comment by kmandrika** [comment]
Hey folks - web support is definitely on our radar. I will keep this FR updated as more concrete details materialize.

**Comment by RoarGronmo** [comment]
@kmandrika It is now in start of september '25, and it has been 5 1/2 mths since we heard from you here. Can you provide some "leaks" from the boards on this issue ?

**Comment by tusharbhambere** [comment]
+1

**Comment by LanderBeeuwsaert** [comment]
My guess/hope is that the new telemetry service is related to this? WDYT? https://github.com/firebase/firebase-js-sdk/pull/9239

**Comment by abdallahshaban557** [comment]
Hi Everybody - this is Abdallah from the Firebase team. We're investigating adding web observability to Crashlytics, and we'd like to get your feedback on our [RFC (Request for comments)](https://github.com/firebase/firebase-js-sdk/discussions/9273). Let us know your thoughts!

**Comment by XeroDays** [comment]
+1

**Comment by umeshpatidarigt** [comment]
+1

**Comment by krystian-dev548** [comment]
+1

**Comment by therajanmaurya** [comment]
+1

**Comment by aronrodrigues** [comment]
+1

**Comment by jwolfson-goog** [comment]
Hi everyone, it's Jenn from Firebase Crashlytics! 

We've been working hard to add support for next.js, Angular, React, and JS apps and are looking for early access testers to try it out and provide feedback. This MVP includes client-side error reporting, de-minified stack traces, and a dedicated web dashboard in the Firebase Console.

Interested? Sign up to test it out here: https://forms.gle/38mjqHGLEZkuksxj8 🔥
